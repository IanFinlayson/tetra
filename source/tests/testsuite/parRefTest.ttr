def main():
   x = [0]
   print("In main(), just set x = [0] and x = ", x, "\n")
   #Test ability to handle many pass by references
   parallel:
      inc(x)
      inc(x)
      inc(x)
      inc(x)
      parInc(x)
      inc(x)
      inc(x)
      inc(x)
      parInc(x)
   print("At the end of main and x[0] = ", x[0], "\n")

def inc(z [int]):
   lock p:
      z[0] += 1

def parInc(z [int]):
   parallel:
      inc(z)
      inc(z)
      inc(z)
   lock p:
      print("here","\n")
      z[0] += 1
