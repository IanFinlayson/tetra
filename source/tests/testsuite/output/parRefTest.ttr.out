indexNum = 0
indexNum = 0
indexNum = 0
indexNum = 0
indexNum = 0
indexNum = 0
indexNum = 0
indexNum = 0
indexNum = 0
indexNum = 0
indexNum = 0
indexNum = 0
indexNum = 0
here
indexNum = 0
here
indexNum = 0
indexNum = 0

